oak-lucene/src/test/java/org/apache/jackrabbit/oak/composite/CompositeNodeStoreLuceneIndexTest.java

```diff
@@ -0,0 +1,506 @@
+
+
+        assertEquals("/content-" + indexedProperty + "/node-0, /content-" + indexedProperty + "/node-1, " +
+        "/libs/node-" + indexName + "-0, /libs/node-" + indexName + "-1, /libs/node-" + indexName + "-2", getResult(result,"jcr:path"));
+
+        // ****This point we have V1 of our app with index properly configured in read only V1 and and global read write****
+    }
+
+    void setupIndexAndContentInReadOnlyV1(String indexName, String indexedProperty, int version) throws Exception {
+        String versionProp = (version == VERSION_1) ? "@v1" : "@v2";
+        // Add some content to read-only v1 repo
+        for (int i = 0; i < 3; i++) {
+            Node b = readOnly1Root.getNode("libs").addNode("node-" + indexName + "-" + i, NT_UNSTRUCTURED);
+            b.setProperty(indexedProperty, "bar");
+            b.setPrimaryType(NT_UNSTRUCTURED);
+        }
+
+        // index creation
+        createLuceneIndex(readOnlyV1Session, indexName, indexedProperty);
+        readOnly1Root.getNode(INDEX_DEFINITIONS_NAME).getNode(indexName).setProperty(IndexConstants.USE_IF_EXISTS, "/libs/indexes/" + indexName + "/@" + versionProp);
+        readOnly1Root.getNode(INDEX_DEFINITIONS_NAME).getNode(indexName).setProperty(FulltextIndexConstants.PROP_REFRESH_DEFN,true);
+
+        // Add /libs/indexes/luceneTest/@v1 to make this index usable/enabled
+        addOrGetNode(readOnly1Root.getNode("libs"), "indexes", NT_UNSTRUCTURED).addNode(indexName, NT_UNSTRUCTURED).setProperty(versionProp.replace("@", ""), true);
+        readOnlyV1Session.save();
+    }
+
+    void setupIndexAndContentInReadOnlyV2(String indexName, String indexedProperty, int version) throws Exception {
+        String versionProp = (version == 1) ? "@v1" : "@v2";
+        // Add some content to read-only v1 repo
+        for (int i = 0; i < 3; i++) {
+            Node b = readOnly2Root.getNode("libs").addNode("node-" + indexName + "-" + i, NT_UNSTRUCTURED);
+            b.setProperty(indexedProperty, "bar");
+            b.setPrimaryType(NT_UNSTRUCTURED);
+        }
+
+        // index creation
+        createLuceneIndex(readOnlyV2Session, indexName, indexedProperty);
+        readOnly2Root.getNode(INDEX_DEFINITIONS_NAME).getNode(indexName).setProperty(IndexConstants.USE_IF_EXISTS, "/libs/indexes/" + indexName + "/@" + versionProp);
+        readOnly2Root.getNode(INDEX_DEFINITIONS_NAME).getNode(indexName).setProperty(FulltextIndexConstants.PROP_REFRESH_DEFN,true);
+
+        // Add /libs/indexes/luceneTest/@v1 to make this index usable/enabled
+        addOrGetNode(readOnly2Root.getNode("libs"), "indexes", NT_UNSTRUCTURED).addNode(indexName, NT_UNSTRUCTURED).setProperty(versionProp.replace("@", ""), true);
+        readOnlyV2Session.save();
+    }
+
+    private String getResult(QueryResult result, String propertyName) throws RepositoryException {
```

Comment:
I think this could be static... could you make all such helper methods static please?

Reply:
done